File: core/src/main/java/com/ataulm/chunks/repository/GsonItemConverter.java
package com.ataulm.chunks.repository;

import com.ataulm.chunks.Item;
import com.ataulm.chunks.Id;

public class GsonItemConverter {

    public GsonItem convert(Item item) {
        GsonItem gsonItem = new GsonItem();
        gsonItem.id = String.valueOf(item.id().value());
        gsonItem.value = item.value();
        gsonItem.completedTimestamp = item.completedTimestamp().isPresent() ? item.completedTimestamp().get() : null;
        return gsonItem;
    }

    public Item convert(GsonItem gsonItem) {
        return Item.createFrom(
                Id.createFrom(gsonItem.id),
                gsonItem.value,
                gsonItem.completedTimestamp
        );
    }

}

File: core/src/main/java/com/ataulm/chunks/repository/GsonItemConverter.kt
package com.ataulm.chunks.repository

import com.ataulm.chunks.Id
import com.ataulm.chunks.Item

class GsonItemConverter {

    fun convert(item: Item): GsonItem {
        val gsonItem = GsonItem()
        gsonItem.id = item.id.value.toString()
        gsonItem.value = item.value
        gsonItem.completedTimestamp = item.completedTimestamp
        return gsonItem
    }

    fun convert(gsonItem: GsonItem): Item {
        return Item.createFrom(
                Id.createFrom(gsonItem.id),
                gsonItem.value,
                gsonItem.completedTimestamp
        )
    }

}
